state/interface_test.go

```diff
@@ -40,7 +40,10 @@ var (
 	_ AgentEntity = (*Machine)(nil)
 	_ AgentEntity = (*Unit)(nil)
 
-	_ ModelAccessor = (*State)(nil)
+	_ ModelAccessor = (*struct {
+		*State
```

Comment:
ModelAccessor should with this refactoring just be implemented by Model, not State